### DISCUSSION TITLE
Query String : rewrite rules and translations

### DISCUSSION THREAD

**Discussion by tonyyb** [livewire/livewire#1252] [4. General]
Hi @calebporzio, first of all thanks for your job : Livewire + Alpine JS is a wonderful combo !

My question is about Query String.

Usage : 
Facets search stuff.

Observations : 
1° For multi-languages applications, query keys must be translated ;
2° Array parameters like : ?search=john&brand[0]=1&brand[1]=2 are not URL friendly.

What do you think about :  `search=john&brand=1,2`  and in french `mots-cles=john&marque=1,2` ?

But today it's quite difficult to get this because of : 
* Rewrite rules are not implemented natively ;
* `history.replaceState` is used on frontend with public keys names ;

Are these points in V2 scope or would you find it interesting to develop these points?

Best regards.

**Comment by calebporzio** [comment]
Interesting, v2 will use pushState. But I don’t think there’s a good way to do translation unfortunately. It’s at least  not in the plans.

**Reply by tonyyb** [reply]
Thanks for your reply.

If you want I can wait for V2 and later make a PR or a gist to discuss about a way to do that (translated parameters).
